src/Mimi.jl

```diff
@@ -216,14 +216,15 @@ function setparameter(m::Model, component::Symbol, name::Symbol, value)
 end
 
 function checklabels(m::Model, component::Symbol, name::Symbol, parametername::Symbol, p::ArrayModelParameter)
-    if !(eltype(p.values) <: getmetainfo(m, component).parameters[parametername].datatype)
+    meta_comp_name = Symbol(string(m.components2[component].component_type))
+    if !(eltype(p.values) <: getmetainfo(m, meta_comp_name).parameters[parametername].datatype)
```

Comment:
Actually, there is also a bug in ``getmetainfo``. It should be ``meta_component_name = Symbol(m.components2[componentname].component_type.name.name)`` there, not what it is right now.

And then you'll also have to fix the other calls to ``getametainfo`` in ``checklabels``.